sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py

```diff
@@ -186,7 +185,9 @@ def __init__(
 
         is_spark_component = isinstance(component, SparkComponent)
         if is_spark_component:
-            self.conf = self.conf or component.conf
+            # conf is dict and we need copy component conf here, otherwise node conf setting will affect component
+            # setting
+            self.conf = self.conf or copy.copy(component.conf)
```

Comment:
This is a little confusing. Let's clarify the expected behavior - if conf is not passed, we want self.conf a *reference* or a *copy* or a *const reference* of self.component.conf?

Reply:
a copy of self.component.conf, because node level conf will be updated like this:
        if self.executor_instances is None and str(self.dynamic_allocation_enabled).lower() == "true":
            self.executor_instances = self.dynamic_allocation_min_executors

But I think it's not expected to change component behavior, because it's confusing that registered component is not as expected in yaml.